Description: Re-introduce fake xs_restrict API call
 libxenstore cannot remove an API function without changing its version
 number. As long as we want to remain with 3.0 we have to keep it around.
 Debian might decide to increment the version at some point but we do not
 know how and when. So for now keep the version stable.
Author: Stefan Bader <stefan.bader@canonical.com>
Forwarded: No

--- a/tools/xenstore/include/xenstore.h
+++ b/tools/xenstore/include/xenstore.h
@@ -133,6 +133,11 @@ bool xs_mkdir(struct xs_handle *h, xs_tr
 bool xs_rm(struct xs_handle *h, xs_transaction_t t,
 	   const char *path);
 
+/* Fake function which will always return false (required to let
+ * libxenstore remain at 3.0 version.
+ */
+bool xs_restrict(struct xs_handle *h, unsigned domid);
+
 /* Get permissions of node (first element is owner, first perms is "other").
  * Returns malloced array, or NULL: call free() after use.
  */
--- a/tools/xenstore/xs.c
+++ b/tools/xenstore/xs.c
@@ -792,6 +792,12 @@ unwind:
 	return false;
 }
 
+/* Always return false a functionality has been removed in Xen 4.9 */
+bool xs_restrict(struct xs_handle *h, unsigned domid)
+{
+	return false;
+}
+
 /* Watch a node for changes (poll on fd to detect, or call read_watch()).
  * When the node (or any child) changes, fd will become readable.
  * Token is returned when watch is read, to allow matching.
